src/eko/matching_conditions/as2.py

```diff
@@ -74,7 +74,7 @@ def A_qq_ns(n, sx, L):
 
 
 @nb.njit(cache=True)
-def A_hq_ps(n, sx, L):
+def A_hq_ps(n, S2, L):
```

Comment:
I think one of the goal of this PR should be to unify the signatures* - or are you going to do this in a separate step? such that you do not need to know for e.g. `A_hq_ps` I do need exactly `S2` and not anything else ...

*do you actually agree?